### PR TITLE
Bump all dependencies

### PR DESCRIPTION
**Pull request by jesse-c**
For some reason, Dependabot doesn't seem to be creating pull requests. This manually updates them for now.

Contributes to PLA-125.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN apt-get update
 
 # Install pip and poetry
 RUN pip install --upgrade pip
-RUN pip install "poetry==1.2.2"
+RUN pip install "poetry==1.8.2"
```
Comment (THOR300) [inline Dockerfile:11]: There is a later version now I've got 1.8.3 locally.

Comment (THOR300) [inline Dockerfile:11]: There is a later version now I've got 1.8.3 locally.

File: pyproject.toml [hunk 2]
```diff
@@ -5,28 +5,28 @@ description = ""
...
+click = "^8.1.7"
+pytest = "^8.3.2"
+pydantic = "^2.8.2"
+python-json-logger = "^2.0.7"
+awscli = "^1.34.10"
+boto3 = "^1.35.10"
 aws-error-utils = "^2.7.0"
-moto = "^4.1.11"
-torch = "2.0.0"
+moto = "^5.0.13"
```
Comment (THOR300) [inline pyproject.toml:18]: Major but a testing library so not core functionality (thumbs up)

**Review by THOR300 — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (jesse-c) [reply Dockerfile]: ✅

File: pyproject.toml (hunk 2 above)
Comment (jesse-c) [reply pyproject.toml:18]: 👌🏻

**Pull request merged**